win32ss/drivers/videoprt/resource.c

```diff
@@ -208,15 +208,16 @@ IntVideoPortReleaseResources(
     BOOLEAN ConflictDetected;
     // An empty CM_RESOURCE_LIST
     UCHAR EmptyResourceList[FIELD_OFFSET(CM_RESOURCE_LIST, List)] = {0};
-
-    Status = IoReportResourceForDetection(
-                DeviceExtension->DriverObject,
-                NULL, 0, /* Driver List */
-                DeviceExtension->PhysicalDeviceObject,
-                (PCM_RESOURCE_LIST)EmptyResourceList,
-                sizeof(EmptyResourceList),
-                &ConflictDetected);
-
+    if (DeviceExtension->IsLegacyDevice)
+    {
+        Status = IoReportResourceForDetection(
+                    DeviceExtension->DriverObject,
+                    NULL, 0, /* Driver List */
+                    DeviceExtension->PhysicalDeviceObject,
+                    (PCM_RESOURCE_LIST)EmptyResourceList,
+                    sizeof(EmptyResourceList),
+                    &ConflictDetected);
+    }
     if (!NT_SUCCESS(Status))
```

Comment:
There's now a bug. If IsLegacyDevice is FALSE (i.e. PNP), the Status variable stays non-initialized.
You should either set Status to some default, or move that if-block inside the ` if (DeviceExtension->IsLegacyDevice)` one.
Or, have an `else` block where you set Status to some default, as you've done in the other function.